containers/src/Kokkos_UnorderedMap.hpp

```diff
@@ -119,6 +119,36 @@ class UnorderedMapInsertResult {
   uint32_t m_status;
 };
 
+/// \class UnorderedMapInsertOpTypes
+///
+/// \brief Operations applied to the values array upon subsequent insertions.
+///
+/// The default behavior when a k,v pair already exists in the UnorderedMap is
+/// to perform no operation. Alternatively, the caller may select to
+/// instantiate the UnorderedMap with the AtomicAdd insert operator such that
+/// duplicate keys accumulate values into the given values array entry.
+/// \tparam ValueTypeView The UnorderedMap value array type.
+/// \tparam ValuesIdxType The index type for lookups in the value array.
+///
+/// Supported operations:
+///   NoOp:      the first key inserted stores the associated value.
+///   AtomicAdd: duplicate key insertions sum values together.
+template <class ValueTypeView = int, class ValuesIdxType = uint32_t>
+struct UnorderedMapInsertOpTypes {
+  using value_type = typename ValueTypeView::non_const_value_type;
+  struct NoOp {
+    KOKKOS_FUNCTION
+    void op(ValueTypeView, ValuesIdxType, const value_type *) const {}
```

Comment:
Why did you decide to pass a pointer to const rather than passing by value for the last argument?
What types to you expect `value_type` might be?  It is a pessimization for arithmetic types.

Reply:
The signatures of `op` must match.

Reply:
So?  They both could take a value.  The comment applies to both.

Reply:
Agreed there is no reason these guys take the value as a pointer

Reply:
Thinking about this more now, a pointer was used to work around compiler errors when value_type = void for non-insertable UnorderdMaps.